comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent_test.go

```diff
@@ -313,6 +313,13 @@ func TestFetchSecurityAgent(t *testing.T) {
 		fetchSecurityConfig = configFetcher.SecurityAgentConfig
 	}()
 	fetchSecurityConfig = func(config pkgconfigmodel.Reader) (string, error) {
+		// test that the agent config was passed and not the system-probe config.
+		assert.False(
+			t,
+			config.IsSet("system_probe_config.sysprobe_socket"),
+			"wrong configuration received for security-agent fetcher",
+		)
+
```

Comment:
Should we add another assertion checking the a value for the agent config is present? 

Same for the ones below 😄